### PR TITLE
Fixing offline table update

### PR DESCRIPTION
**Pull request by japoorv**
Fixing offline table update.

### CODE REVIEW

**Review by laxmanchekka — CHANGES_REQUESTED** [review]

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 1]
```diff
@@ -368,29 +376,90 @@ public static boolean sendPinotTableCreationRequest(
...
+        LOGGER.error("Failed to update Pinot table.", e);
+        return false;
+      }
     } catch (Exception e) {
       LOGGER.error("Failed to create Pinot table.", e);
       return false;
     }
   }
 
+  public static String sendRequest(String requestMethod, String urlString, String payload)
```
[the remark this answers is not in the document]
Comment (japoorv) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:414]: We are using the return String to log.

**Review by laxmanchekka — DISMISSED** [review]

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 2]
```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
...
+      conn.setRequestMethod(requestMethod);
+
+      final BufferedWriter writer =
+          new BufferedWriter(
+              new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+      writer.write(payload, 0, payload.length());
+      writer.flush();
+
+      final int responseCode = conn.getResponseCode();
+      if (responseCode != 200) {
```
Comment (rish691) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:421]: static variable for the response code 200, also use the same while returning

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 3]
```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
...
   public static boolean sendPinotTableCreationRequest(
-      String controllerHost, String controllerPort, final String tableConfig) {
+      String controllerHost,
+      String controllerPort,
+      final String tableConfig,
+      final String tableName) {
+    String controllerAddress = getControllerAddressForTableCreate(controllerHost, controllerPort);
+    LOGGER.info("Trying to send table creation request {} to {}. ", tableConfig, controllerAddress);
+    int responseCode = sendRequest("POST", controllerAddress, tableConfig);
+    if (responseCode == 409) {
```
Comment (rish691) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:389]: static var for response code 409

Comment (tim-mwangi) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:389]: Some comment around what this means and how we proceed

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 4]
```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
...
+    return true;
+  }
+
+  /**
+   * @param requestMethod: Type of request (PUT/GET/POST/DELTE)
+   * @param urlString: Api endpoint
+   * @param payload: Payload
+   * @return Status Code
+   */
+  public static int sendRequest(String requestMethod, String urlString, String payload) {
```
Comment (tim-mwangi) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:405]: Do we have to impl this ourselves? Nothing in the libraries to help?

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 2 above)
Comment (japoorv) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:421]: Okay

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 3 above)
Comment (japoorv) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:389]: Okay

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 4 above)
Comment (japoorv) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:405]: Actually previously, we were using Pinot library for creating/updating library. But there implementation is a bit buggy in the sense that we can't update OFFLINE tables (though REALTIME can be) via POST request to api. So we are explicitly initiating the request.

Comment (tim-mwangi) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:412]: ok

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**